src/OutputHash.js

```diff
@@ -56,26 +56,85 @@ function replaceStringInAsset(asset, source, target) {
  */
 function reHashChunk(chunk, assets, hashFn) {
     const oldHash = chunk.renderedHash;
-    const oldChunkName = chunk.files[0];
+    const fileIndex = chunk.files.findIndex(file => file.endsWith('.js'));
+    if (fileIndex === -1) {
+        return {};
+    }
+
+    const oldChunkName = chunk.files[fileIndex];
     const asset = assets[oldChunkName];
     const { fullHash, shortHash } = hashFn(asset.source());
     const newChunkName = oldChunkName.replace(oldHash, shortHash);
 
     // Update the main file of the chunk with the new name
     chunk.hash = fullHash;
     chunk.renderedHash = shortHash;
-    chunk.files[0] = newChunkName;
+    chunk.files[fileIndex] = newChunkName;
 
     // Update the asset associated with that file
     asset._name = newChunkName;
     delete assets[oldChunkName];
     assets[newChunkName] = asset;
 
     // Update the content of the rest of the files in the chunk
-    chunk.files.slice(1).forEach((file) => {
-        const secondaryAsset = assets[file];
-        replaceStringInAsset(secondaryAsset, oldHash, shortHash);
-    });
+    chunk.files
+        .filter(file => file !== newChunkName)
+        .forEach((file) => {
+            const secondaryAsset = assets[file];
+            replaceStringInAsset(secondaryAsset, oldHash, shortHash);
+        });
+
+    return {
+        oldHash,
+        newHash: shortHash,
+    };
+}
+
+function getMiniCssContentHashKey(chunk) {
+    return Object.keys(chunk.contentHash).find(hashType => hashType.includes('/mini-css-extract-plugin/'));
+}
+
+/**
+ * Computes the new hash of a CssModule.
+ *
+ * This function updates the *name* of the main file (i.e. source code), and the *content* of the
+ * secondary files (i.e source maps)
+ */
+function reHashCssAsset(chunk, assets, hashFn) {
```

Comment:
Lots of similar code to `reHashChunk` in here.  Ideally, I'd be able to refactor this in a follow-up PR.